README.md

```diff
@@ -1,3 +1,134 @@
+generated between any two requested timestamps. Motion model constraints are generated upon request, not due to their
+own internal clock. We use the motion model to connect the states introduced by the sensor measurements. So we derive
+a class from the `fuse_core::MotionModel` base class that implements our differential drive kinematic constraints.
+
+![fuse graph](doc/fuse_graph_3.png)
+
+Note that there is nothing preventing the system from using multiple motion models at the same time, if that is of
+benefit.
+
+The two sensor models and the motion model are configured as plugins for an optimizer implementation. The optimizer
+handles all of the coordination between the sensors and motion models, as well as performs the required computation
+to generate the optimal state variable values.
+
+![fuse graph](doc/fuse_graph_4.png)
+
+While our `fuse` system is now optimizing constraints from two different sensors and a motion model, it is not yet
+publishing any data back out to ROS. We should probably do that. So we would derive a `fuse_core::Publisher` class
+that finds the most recent position and orientation and publishes that as a `geometry_msgs::PoseStamped` message to
+the topic of your choice.
+
+![fuse graph](doc/fuse_graph_5.png)
+
+We finally have something that is starting to be useful.
+
+### Adaptation #1: Full path publishing
+
+Nothing about the `fuse` framework limits you to having a single publisher. What if you want to visualize the entire
+robot trajectory, instead of just the most recent pose? Well, we can create a new derived `fuse_core::Publisher` class
+that publishes all of the robot poses using a `nav_msgs::Path` message.
+
+![fuse graph](doc/fuse_graph_adaptation1.png)
+
+### Adaptation #2: Changing kinematics
+
+In your spare time, you also build [autonomous power wheels racers](http://www.powerracingseries.org/). But race cars
+don't use differential drive; you need a different motion model. Easy enough. We simply derive a new
+`fuse_core::MotionModel` class that implements an Ackermann steering model. Everything else can be reused.
+
+![fuse graph](doc/fuse_graph_adaptation2.png)
+
+### Adaptation #3: Online calibration
+
+Over time you notice that the accuracy of the odometry measurements is decreasing. After some investigation you realize
+that the soft rubber racing tires are wearing, decreasing the diameter of the wheels over time. It sure would be nice
+if the odometry system could compensate for that automatically. To do that, we will create a new variable type, a wheel
+diameter. We will also need a new wheel encoder sensor model. This new model will involve the previous pose and
+next pose as before, but it will also involve the previous wheel diameter estimate. Finally, we will need to create
+a "motion model" that describes how the wheel diameter is expected to change over time. Maybe some sort of exponential
+decay?
```

Comment:
I would throw some class names in here, i.e. The wheel diameter is a new Variable, etc.